### PR TITLE
Deprecating i/u suffixes in libcoretest

### PR DESCRIPTION
**Pull request by alfiedotwtf**

### CODE REVIEW

File: src/libcoretest/any.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ static TEST: &'static str = "Test";
 
 #[test]
 fn any_referenced() {
-    let (a, b, c) = (&5u as &Any, &TEST as &Any, &Test as &Any);
+    let (a, b, c) = (&5 as &Any, &TEST as &Any, &Test as &Any);
 
     assert!(a.is::<uint>());
```
Comment (tbu-) [inline src/libcoretest/any.rs]: That needs more fixing.

**Comment by alfie** [comment]
@tbu- could you please elaborate?

File: src/libcoretest/any.rs (hunk 1 above)
Comment (tbu-) [inline src/libcoretest/any.rs]: This makes the the literal a `u32`, but later it's checked if it's a `u32`.

**Comment by alfie** [comment]
I think type inference knows it's a uint as 'make check' passed all tests.

**Comment by tbu-** [comment]
Let me claim that this is impossible or a bug in the type inference.

**Comment by tbu-** [comment]
`a.rs`:

    use std::any::Any;
    fn main() {
        println!("{}", (&5 as &Any).is::<uint>())
    }

Prints:

    $ rustc a.rs && ./a
    false

**Comment by tbu-** [comment]
If you use `make -j <anything greater than 1> check`, it can squelch test errors.

**Comment by alfie** [comment]
What's the default, as I didn't use -j?

**Comment by tbu-** [comment]
Default should be fine. But have you seen my code snippet that demonstrates that this test cannot succeed?

**Comment by alfie** [comment]
Yeah, you're definitely correct. I'm was just worried about how my tests were passing. And if so, there must be other commits out there that aren't being caught too.

I'll fix my change... but why did make check pass?!

**Comment by alfie** [comment]
Looking at the output, there is zero mention of libcoretest. So I guess these tests aren't even run with 'make check'?

**Comment by tbu-** [comment]
Mh. They should be.

**Comment by alfie** [comment]
Oh, this explains so much :(

  cfg: antlr4 not available, skipping lexer test...
  cfg: flex not available, skipping parser test...

Installing and running tests again. Sorry for the run around.

**Comment by alfie** [comment]
Well this is weird. libcoretests are definately not firing on my local machine. I've made a post on internal:

  http://internals.rust-lang.org/t/make-check-not-firing-libcoretest-tests/1534

**Comment by tbu-** [comment]
@alfie These error messages should have nothing to do with `libcoretest`. You could try `make check-coretest-stage1`.

**Comment by alfie** [comment]
Yep. I upgraded Valgrind and now something completely unrelated is failing:

    test old_io::net::tcp::test::bind_error ... FAILED
    test old_io::net::udp::test::bind_error ... FAILED

So back to the drawing board - now that I have all dependencies (flex, bison, antlr4 and valgrind were missing or old), I'm currently running 'make check' on upstream's master to see if there's anything else wrong with my local machine before continuing with this PR.

**Comment by alfie** [comment]
@tbu- sorry about the delay. Worked out everything I needed, and was able to identify all the issues. Make check happy now, so this PR is good to go.

File: src/libcoretest/any.rs [hunk 2]
```diff
@@ -18,7 +18,7 @@ static TEST: &'static str = "Test";
 
 #[test]
 fn any_referenced() {
-    let (a, b, c) = (&5u as &Any, &TEST as &Any, &Test as &Any);
+    let (a, b, c) = (&5us as &Any, &TEST as &Any, &Test as &Any);
```
Comment (tbu-) [inline src/libcoretest/any.rs]: This should probably be changed to use `i32`, there's no need for a platform dependent integer type here.

**Comment by tbu-** [comment]
Except for the one note I left, this looks fine.

**Comment by alfie** [comment]
Thanks for the suggestion, but I tried modifying as you said (5us to 5i32) and it failed:

    failures:
    
    ---- any::any_referenced stdout ----
        thread 'any::any_referenced' panicked at 'assertion failed: a.is::<uint>()', /home/alfie/rust/src/libcoretest/any.rs:23

So unless you have any other suggestions, 1f00690 was the last one that passed all tests.

**Comment by tbu-** [comment]
@alfie 
    #[test]
    fn any_referenced() {
        let (a, b, c) = (&5i32 as &Any, &TEST as &Any, &Test as &Any);
        assert!(a.is::<i32>());
        assert!(!b.is::<i32>());
        assert!(!c.is::<i32>());
        assert!(!a.is::<&'static str>());
        assert!(b.is::<&'static str>());
        assert!(!c.is::<&'static str>());
        assert!(!a.is::<Test>());
        assert!(!b.is::<Test>());
        assert!(c.is::<Test>());
    }

I.e. change all the checks for `uint` to checks for `i32`, too. Thanks for your endurance!

**Comment by alfie** [comment]
@tbu- Ahh I see. I only wanted the bare minimum (so only changing the suffixes), but that one makes sense. Fixed now. Thanks!

**Comment by alfie** [comment]
ping @tbu-

**Comment by alfie** [comment]
Awesome. Thanks @pnkfelix and @tbu-!

**Comment by tbu-** [comment]
Great to see it landed. Thanks for your contribution. :)

**Pull request merged**